### DISCUSSION TITLE
Cargo build failes to parse manifest Cargo.toml

### DISCUSSION THREAD

**Discussion by morlandi** [dani-garcia/vaultwarden#2091] [Q&A] [answered]
While trying to issue this command:

```
cargo build --release --features sqlite
```

I receive the following error message:

```
error: failed to parse manifest at `/home/vault/vaultwarden/Cargo.toml`

Caused by:
  feature `edition2021` is required

  consider adding `cargo-features = ["edition2021"]` to the manifest
```

Environment:

```
Ubuntu 20.04.3 LTS (GNU/Linux 5.4.0-86-generic x86_64)

$ cargo --version
cargo 1.53.0
```

Since I'm totally new to rust and Cargo, any help or suggestion would be much appreciated

**Comment by BlackDex** [comment, marked as answer]
You need to use a nightly version v1.57 or above.

**Comment by morlandi** [comment]
Thank you, that worked.

For anyone interested (and newbie to rust like me), that's what I did:

```
$ apt remove rustc
$ curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
$ cargo --version
cargo 1.58.0-nightly (94ca096af 2021-10-29)
```

**Reply by csharpnewbie3** [reply]
hey @morlandi I am using the same command to install but still it is showing as cargo 1.46.0. I have ubuntu version 16.04
